### ISSUE TITLE
FMN Version 3 Beta Online

### ISSUE THREAD

**Issue by Lusito** [Lusito/forget-me-not#253]
It's been very quiet lately. I'm sorry for that. The amount of time I can spend on this project varies and Mozilla lately made a lot of decisions that make me question whether It's worth continuing developing web-extensions for Firefox.

Anyway, let's keep this concise here.
I've uploaded a beta version, which you can try. Keep in mind though, that this has not been tested as extensively in the browser yet, as the amount of features and different scenarios that need to be tested keeps growing and growing with every release.

If possible, use firefox beta, developer or nightly and not your main profile:
https://www.mozilla.org/de/firefox/channel/desktop/

Here's a page with which you can test some things: https://lusito.github.io/forget-me-not/

**Last warning:** Even if I increased the amount of unit-tests to 551, there is still a chance I overlooked something and you might lose data when trying this!

Install the beta by [clicking here](http://lusito.github.io/forget-me-not/forget-me-not-beta.xpi)

**Comment by atomGit** [comment]
> Mozilla lately made a lot of decisions that make me question whether It's worth continuing developing web-extensions for Firefox.

let me guess ... they broke the API's **again**???

i can't blame you, but it's really sad to see capable, ethical developers becoming disillusioned with Mozilla and Firefox - i think quite a few very good add-on devs jumped ship because of the loonies at Moz corporate

**Comment by geeknik** [comment]
Firefox is still leaps and bounds better than the alternative. Is there anything the community can do to help?

**Comment by atomGit** [comment]
i agree - i use and recommend Firefox in my guides (12bytes.org) because i see it as the only viable browser at this time that's well suited to privacy/security hardening, however i have lost a lot of faith in Mozilla over the years as a result of some incredibly idiotic decisions, unethical corporate relationships, and their desire to emulate g.chrome

**Comment by ghost** [comment]
Curious to try this... I wonder, is there an easy transition of settings from an existing installation?

**Comment by Lusito** [comment]
> let me guess ... they broke the API's again???

Well, not exactly, but they broke android support for all but 9 exclusive extensions.

> Firefox is still leaps and bounds better than the alternative.

The important thing here is the word "still". I wonder for how long. If they keep focussing on stuff like rewriting the URL bar every couple of month just to piss of users even more than the last time they changed it, they will lose more users and thus income, which keeps them afloat.

> Is there anything the community can do to help?

- If you JavaScript/TypeScript, you can contribute to the code
- If you know speak multiple languages good enough, you can create/update translations
- Otherwise, I'm afraid the only thing you can do is test the beta extensively.

You can also vote for these issues (login via github is possible):
- https://bugzilla.mozilla.org/show_bug.cgi?id=1340511
- https://bugzilla.mozilla.org/show_bug.cgi?id=1353726
- https://bugzilla.mozilla.org/show_bug.cgi?id=1380445
- https://bugzilla.mozilla.org/show_bug.cgi?id=1440263
- https://bugzilla.mozilla.org/show_bug.cgi?id=1464895
- https://bugzilla.mozilla.org/show_bug.cgi?id=1526246
- https://bugzilla.mozilla.org/show_bug.cgi?id=1633315

> Curious to try this... I wonder, is there an easy transition of settings from an existing installation?

You can export your settings in the Settings->General tab of the old version and then import that file from the new version. Should be compatible if I didn't mess up somewhere.

**Comment by atomGit** [comment]
> You can also vote  ...

well i can't code, but i can vote ... done!

**Comment by ghost** [comment]
FMN 3.0.0beta6 / Firefox 84.0 (x64) / Windows 7 (x64).

`Otherwise, I'm afraid the only thing you can do is test the beta extensively.`

Not extensively yet but first experience is flawless (FMN Tester, AMO, YouTube). Worked as with previous stable versions of FMN, now IndexedDB and others  wiped correctly as well (Caching question though hereafter). No issues up to now. I notice also thet FMN 3.0 weighs 816.4 KB as opposed to ver. 2.2.8 and its 1.89MB

If I encounter any issue of course I'll report it here.

Side-Note : what exactly does _Caching_ include? Be it with _FMN_ as with competitor extension _Cookie Autodelete_, cache cleaning is included but with either extension I don't see a site's cache having been wiped when I exit that site and go to Firefox's about:cache ...

Nice work, thanks.

EDIT: forgot to mention, concerning the Cache cleaning, that Disk cache is here disabled, only memory cache, which maybe explains it all?

**Comment by B00ze64** [comment]
I voted for all but one of the bugs; you are very thorough, finding that cookie/../../ is quite a catch. And don't worry about Android Firefox, they'll eventually unblock all extensions (hopefully CAD is not a "Recommended" extension that gets unblocked before everyone else, hurting FMN somewhat). Besides, who uses Firefox on Android? Personally I use Samsung's browser, funny I should use that one, but the buttons (back/forward/etc) are at the bottom and the URL bar at the top, and nothing else makes as much sense as that arrangement as far as I'm concerned - I get to have a full row of buttons at the bottom (read: many buttons) not limited by the space taken by the URL bar. I sincerely have no idea why no one else does it like that...

**Comment by bege10** [comment]
> 
> 
> > You can also vote  ...
> 
> well i can't code, but i can vote ... done!

for a long time already I don't see the vote feature anymore in bugzilla. Where has it gone?

**Comment by offhub** [comment]
FMN 3.0.0beta6 / Firefox 88.0a1 (x64) / Windows 10 (x64).

Importing doesn't work.

**Comment by atomGit** [comment]
> for a long time already I don't see the vote feature anymore in bugzilla. Where has it gone?

click 'Details'

**Comment by bege10** [comment]
> Well, not exactly, but they broke android support for all but 9 exclusive extensions.

By now there are only a few more 👿 BUT:
You can add your own add-on collection in **Nightly** and in **Fennec** (F-Droid). Firefox for Android Nightly (currently v90.0a) is also available at Play Store, Fennec (= Firefox stable currently v88.0 with add-on support as in Nightly and about:config enabled) available at F-Droid.
The good news is that **FMN works in both these versions!**

If you don't know how to enable the own add-on collection as I did:
- Login to your account in addons.mozilla.org
- Create your own collection (hover over your name at top right and select the first entry)
- Go to FMN page
- Scroll down to bottom of page and add FMN to your collection.
- Go to your account profile and note your user number from URL

.
- Open Nightly or Fennec settings
- Open About at end of the list
- Tap several times on the app icon until debug menu is enabled (similar to enabling developer options in Android)
- Go back to settings and you will find a new entry for user defined add-on collection
- Enter your user number and your collection name
- App will close, reopen and you will find your own list of add-ons
NOTE: Only your own collection will be displayed, not the recommended add-ons. If you want them also you must add them manually to your collection.

I also use a cookie manager that works correct also.
History deleters don't work yet.

**Comment by bege10** [comment]
Export of FMN settings does not work on Android, import does. Is there a permission missing? Other add-ons handle it as a download.
